### ISSUE TITLE
feat request: Make leader key configurable

### ISSUE THREAD

**Issue by mikesmithgh** [roobert/surround-ui.nvim#2]
Hi, I found this on this week in Neovim. I think this is really useful. It took me a little bit to realize the leader key had to be added before the root key. This feature request is to allow the leader key to be configurable:
https://github.com/roobert/surround-ui.nvim/blob/main/lua/surround-ui/init.lua#L52

For example, the default surround config uses y instead of <leader>.

**Comment by linrongbin16** [comment]
Agree, maybe simply set the "root_key" to `<leader>S` and remove the hard coding, so people can configure it if they need other keys to trigger.

**Comment by roobert** [comment]
I think the only way to add this in a backwards-compatible way would be to add another option, something like `disable_leader`. Changing the `root_key` option risks breaking all existing configurations.

Happy to accept a PR for this functionality.
